src/state/pages/actions.js

```diff
@@ -194,6 +194,17 @@ export const fetchPage = wrapApiCall(getPage);
 export const fetchPageInfo = wrapApiCall(SEO_ENABLED ? getPageSEO : getPage);
 export const fetchPageChildren = wrapApiCall(getPageChildren);
 
+export const fetchIfPageExists = pageCode => new Promise((resolve) => {
+  getPage(pageCode).then((response) => {
+    if (response.ok) {
+      resolve(true);
+    } else {
+      resolve(false);
+    }
```

Comment:
maybe here you can resolve directly response.ok

```suggestion
    resolve(response.ok)
```

although I think that if it's not ok it'll go directly to the catch maybe

Reply:
done

Reply:
no it wont go, depends on error code.